File: api/src/main/java/com/ruby/rubia_server/core/entity/QrCodeResult.java
package com.ruby.rubia_server.core.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QrCodeResult {
    private boolean success;
    private Object data;
    private String type;
    private String error;

    public static QrCodeResult success(Object data, String type) {
        return QrCodeResult.builder()
            .success(true)
            .data(data)
            .type(type)
            .build();
    }

    public static QrCodeResult error(String error) {
        return QrCodeResult.builder()
            .success(false)
            .error(error)
            .build();
    }
}

File: api-dotnet/Rubia.Server/Entities/QrCodeResult.cs
namespace Rubia.Server.Entities;

public class QrCodeResult
{
    public bool Success { get; set; }
    public object? Data { get; set; }
    public string? Type { get; set; }
    public string? Error { get; set; }

    public static QrCodeResult CreateSuccess(object data, string type)
    {
        return new QrCodeResult
        {
            Success = true,
            Data = data,
            Type = type
        };
    }

    public static QrCodeResult CreateError(string error)
    {
        return new QrCodeResult
        {
            Success = false,
            Error = error
        };
    }
}